src/main.rs

```diff
@@ -67,7 +75,18 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    let mut count = 0;
+    let mut current_popn = starting_rabbits;
+    while current_popn != 1{
+        if current_popn % 2 == 0{
+            current_popn = current_popn / 2;
+            count += 1;
+        }else{
+            current_popn = (3 * current_popn) + 1;
+            count += 1;
+        }
+    }
+    return count;
```

Comment:
In Rust, when returning from a function, we don't actually need to write the return keyword like `return x;` if it's the last expression. Instead, we can just do `x`. For example:

```rust
fn foo() -> i32 {
    let x = 5;
    x
}
```

Or, to simplify that more:

```rust
fn foo() -> i32 {
    5
}
```

Here are some more docs on that: https://doc.rust-lang.org/rust-by-example/fn.html

Reply:
oh I see. I'm just used to python, c++ functions. Thanks for pointing it out!